src/analyze/analyze.c

```diff
@@ -2423,6 +2426,16 @@ static int parse_argv(int argc, char *argv[]) {
 
                         break;
 
+                case 'U':
+                        if (!unit_name_is_valid(optarg, UNIT_NAME_ANY))
+                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to parse unit name: %s", optarg);
```

Comment:
Use unit_name_mangle() here? like we use everywhere else?

Reply:
Didn't know this existed. Sweet!